core/src/main/java/org/renjin/primitives/io/serialization/RDataReader.java

```diff
@@ -510,7 +510,7 @@ private SEXP readEnv(int flags) throws IOException {
     for (int i = 0; i < hashtab.length(); ++i) {
       PairList node = hashtab.getElementAsSEXP(i);
       while (node != Null.INSTANCE) {
-        env.setVariable(node.getTag(), ((PairList.Node)node).getValue());
+        env.setVariable((Context) readContext, node.getTag(), ((PairList.Node)node).getValue());

```

Comment:
`readContext` is **not** an instance of org.renjin.eval.Context. This is is case where you could use the EnvironmentBuilder to initialize the Environment without having to worry if there are active bindings...